src/AlgebraicGeometry/ToricVarieties/AlgebraicCycles/special_attributes.jl

```diff
@@ -101,11 +101,11 @@ julia> length(map_gens_of_chow_ring_to_cox_ring(p2))
 @attr Dict{QQMPolyRingElem, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}} function map_gens_of_chow_ring_to_cox_ring(v::AbstractNormalToricVariety)
     g = gens(chow_ring(v))
     g2 = gens(cox_ring(v))
-    r_list = [rays(c) for c in cones(v)]
+    cs = cones(v)
     mapping = Dict{QQMPolyRingElem, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}()
-    for rs in r_list
-      p1 = prod([g[findfirst(x->x==r, rays(v))] for r in rs]).f
-      p2 = prod([g2[findfirst(x->x==r, rays(v))] for r in rs])
+    for k in 1:nrows(cs)
+      p1 = prod([g[l]^(cs[k,l]) for l in 1:nrays(v)]).f
+      p2 = prod([g2[l]^(cs[k,l]) for l in 1:nrays(v)])
```

Comment:
As above, please avoid using `prod` and construct the monomials directly. Then there will be no multiplication which for this step should be much faster. Probably not so relevant for the overall performance.